meson.build

```diff
@@ -34,6 +34,8 @@ if get_option('platform') == 'tizen'
   if get_option('enable-tizen-feature-check')
     add_project_arguments('-D__FEATURE_CHECK_SUPPORT__', language: ['c', 'cpp'])
   endif
+else
+  add_project_arguments('-DENABLE_JSON=1', language: ['c', 'cpp'])
```

Comment:
You prepared this for Android only? Then, add ```elif android``` here.

Ubuntu build:
```
[369/506] Compiling C++ object 'Applications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe/main.cpp.o'.
FAILED: Applications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe/main.cpp.o 
c++ -IApplications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe -IApplications/PicoGPT/jni -I../Applications/PicoGPT/jni -I../Applications/utils/jni/./includes -Inntrainer -I../nntrainer -Inntrainer/../api -I../nntrainer/../api -I../nntrainer/../api/ccapi/include -Inntrainer/compiler -I../nntrainer/compiler -Inntrainer/dataset -I../nntrainer/dataset -Inntrainer/layers/loss -I../nntrainer/layers/loss -Inntrainer/layers -I../nntrainer/layers -Inntrainer/models -I../nntrainer/models -Inntrainer/optimizers -I../nntrainer/optimizers -Inntrainer/tensor -I../nntrainer/tensor -Inntrainer/utils -I../nntrainer/utils -Inntrainer/graph -I../nntrainer/graph -I../api/ccapi/include -Iapi/ccapi/.. -I../api/ccapi/.. -I/usr/include/iniparser -I/usr/include/x86_64-linux-gnu/openblas-pthread/ -I/usr/include/nnstreamer -I/usr/include/tensorflow2/ -I/usr/lib/x86_64-linux-gnu/pkgconfig/../../include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=c++17 '-DVERSION="0.5.0"' -DVERSION_MAJOR=0 -DVERSION_MINOR=5 -DVERSION_MICRO=0 -DENABLE_JSON=1 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wvla -Wpointer-arith -Wno-error=varargs -ftree-vectorize -Wno-maybe-uninitialized -Wno-unused-variable -DMIN_CPP_VERSION=201703L -DOPENCL_KERNEL_PATH=nntrainer_opencl_kernels -DML_API_COMMON=1 -DNNSTREAMER_AVAILABLE=1 -DUSE_BLAS=1 -DNNTR_NUM_THREADS=1 -DOMP_NUM_THREADS=1 -DDEBUG=1 -D__LOGGING__=1 -DENABLE_TEST=1 -DENABLE_NNSTREAMER_BACKBONE=1 -DENABLE_TFLITE_BACKBONE=1 -DENABLE_TFLITE_INTERPRETER=1 '-DNNTRAINER_CONF_PATH="/etc/nntrainer.ini"' -g -O2 -fdebug-prefix-map=/var/www/html/nntrainer/TAOS-CI/ci/repo-workers/pr-checker/2627-202406101303480.65484309196472-85cc5e3160ee72c0ab0bf4fb87950237a52136bb/nntrainer=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DGTEST_HAS_PTHREAD=1 -Wl,--start-group -lpthread -Wl,--end-group -fopenmp -pthread -MD -MQ 'Applications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe/main.cpp.o' -MF 'Applications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe/main.cpp.o.d' -o 'Applications/PicoGPT/jni/4c5dc94@@nntrainer_pico_gpt@exe/main.cpp.o' -c ../Applications/PicoGPT/jni/main.cpp
../Applications/PicoGPT/jni/main.cpp:21:10: fatal error: encoder.hpp: No such file or directory
   21 | #include "encoder.hpp"
      |          ^~~~~~~~~~~~~
compilation terminated.
```

Reply:
@myungjoo Thanks, it seems like this code currently only works on Android well. 

The reason why the CI passed for both Android and Ubuntu before is because, as mentioned in the PR message, the extra_defines related code has never worked properly.

I will only enable this option for Android until additional update measures are taken to make this code work in a general Ubuntu environment(In some cases, such as in my Ubuntu environment, it works, but in other cases, it doesn't, so an update is required.).

I will leave the issue about this. thank you!

Reply:
In issue #2632 , I left the issue of supporting encoders in Ubuntu and Tizen